akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -576,7 +576,7 @@ object Uri {
     }
     def toMultiMap: Map[String, List[String]] = {
       @tailrec def append(map: Map[String, List[String]], q: Query): Map[String, List[String]] =
-        if (q.isEmpty) map else append(map.updated(q.key, q.value :: map.getOrElse(q.key, Nil)), q.tail)
+        if (q.isEmpty) map else append(map.updated(q.key, map.getOrElse(q.key, Nil) :+ q.value), q.tail)
```

Comment:
The only other solution (under the constraint that we cannot change the signature of the method) would be to collect in reverse order as before and then change all values afterwards. So, I agree that it is quite unlikely that people will use it in a way where it starts to matter. If that's the case, they could still provide their own faster implementation.